examples/python/multiprocessing/README.md

```diff
@@ -0,0 +1,68 @@
+## Multiprocessing with gRPC Python
+
+Multiprocessing allows application developers to sidestep the Python global
+interpreter lock and achieve true concurrency on multicore systems.
+Unfortunately, using multiprocessing and gRPC Python is not yet as simple as
+instantiating your server with a `futures.ProcessPoolExecutor`.
+
+The library is implemented as a C extension, maintaining much of the state that
+drives the system in native code. As such, upon calling
+[`fork`](http://man7.org/linux/man-pages/man2/fork.2.html), much of the
+state copied into the child process is invalid, leading to hangs and crashes.
+
+However, calling `fork` without `exec` in your python process is supported
+*before* any gRPC servers have been instantiated. Application developers can
+take advantage of this to parallelize their CPU-intensive operations.
+
+## Calculating Prime Numbers with Multiple Processes
+
+This example calculates the first 10,000 prime numbers as an RPC. We instantiate
+one server per subprocess, balancing requests between the servers using the
+[`SO_REUSEPORT`](https://lwn.net/Articles/542629/) socket option. Note that this
+option is not available in `manylinux1` distributions, which are, as of the time
+of writing, the only gRPC Python wheels available on PyPi. To take advantage of this
+feature, you'll need to build from source, either using bazel (as we do for
+these examples) or via pip, using `pip install grpcio --no-binary grpcio`.
+
+```python
+_PROCESS_COUNT = multiprocessing.cpu_count()
+```
+
+On the server side, we detect the number of CPUs available on the system and
+spawn exactly that many child processes. If we spin up fewer, we won't be taking
+full advantage of the hardware resources available. If we spin up more, then the
+kernel will have to multiplex the processes on the available CPUs.
```

Comment:
Just curious: Is the resource usage confirmed by experiments? Will 2 processes be 2 times faster than single process version?

Reply:
I upped the max to 100,000 to make the effect of fixed costs smaller. On my 12-core machine, I'm seeing:

### Thread Count = Process Count
| Processes | Seconds |
|---------------|-------------|
| 1 | 53.7 |
| 2 | 31.1 |
| 4 | 20.0 |
| 8 | 13.7 |
| 16 | 18.2 |
| 32 | 14.0 |
| 64 | 11.8 |
| 128 | 12.0 |

### Thread Count Fixed to 12
| Processes | Seconds |
|---------------|-------------|
| 1 | 81.0 |
| 2 | 33.4 |
| 4 | 16.7 |
| 8 | 13.7 |
| 16 | 14.5 |
| 32 | 11.3 |
| 64 | 11.4 |
| 128 | 11.3 |

1 process through 16 processes makes sense to me. 32 and beyond... is a bit of a head scratcher...